### DISCUSSION TITLE
Is it possible to listening to all properties' INotifyPropertyChanged

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#360] [Q&A] [answered]
Say I have a class that has bunch of properties recording the program states.  I want have this class serialized to XML and save it when any of its properties changed.  I could do the following

```
var state = new State();
state.Bind(x => x.Prop1, (sender, eventArgs) => SerializeAndSave());
state.Bind(x => x.Prop2, (sender, eventArgs) => SerializeAndSave());
...
state.Bind(x => x.PropN, (sender, eventArgs) => SerializeAndSave());
```

Or automate that with a loop statement (with reflection).  But is there are more elegant way to do so, for example, if I omit the first delegate, it'll default to bind all PropertyChanged event handler to `SerializeAndSave`
```
var state = new State();
state.Bind((sender, eventArgs) => SerializeAndSave());
```

**Comment by canton7** [comment, marked as answer]
Why not just subscribe to the PropertyChanged event directly?

**Reply by mwdpb** [reply]
Brilliant!
